sympy/calculus/util.py

```diff
@@ -85,19 +86,11 @@ def continuous_domain(f, symbol, domain):
         domain = constrained_interval
 
     try:
-        if f.has(Abs):
-            sings = solveset(1/f, symbol, domain) + \
-                solveset(denom(together(f)), symbol, domain)
-        else:
-            for atom in f.atoms(Pow):
-                predicate, denomin = _has_rational_power(atom, symbol)
-                if predicate and denomin == 2:
-                    sings = solveset(1/f, symbol, domain) +\
-                        solveset(denom(together(f)), symbol, domain)
-                    break
-            else:
-                sings = Intersection(solveset(1/f, symbol), domain) + \
-                    solveset(denom(together(f)), symbol, domain)
+        sings = EmptySet()
+        for atom in f.rewrite([sec, csc, cot, tan], cos).atoms(Pow):
+            base, exponent = atom.args
```

Comment:
or `base, exponent = atom.as_base_exp()`

Reply:
I suspect you should simply put this rewriting at the top of the `try` that you deleted and allow it to work as it did before.

Reply:
I've replaced this with `as_base_exp` now (and done all the squashing and testing). The old code would not work with the addition of this rewriting, as it: firstly doesn't look at fractions which are nested inside other expressions (as in sometime like `sin(1/x)` and secondly using `solveset` on `1/f(x)` for an `f(x)` that isn't a fraction will never result in a solution, due to `solveset` not working in any situation where it has to do algebra with infinites. This issue is reported at #18857, but it doesn't look like that's going to be resolved anytime soon, so I thought I would use this alternative method

Reply:
Hmm...I agree with you. What was there for finding singularities does not look like it would work well.

It looks like there are two steps: find the domain and then look for singularities in the domain. I'm not sure why `Abs` was targetted since that is a "absolute continuity" idea. And it's not clear from the docstring which is intended. And no tests are failing without that code. e.g. `sqrt(x - abs(x -3))` will have a domain limited to non-negative args for the `sqrt`; in addition there will be a break in differentiability (but not continuity) at `x = 3`.

Reply:
Looking at other functions in the file, however, I see `function_range`. The only thing that would differentiate these two functions is if the `continuous_domain` considered differentiability in which case it should be taking `Abs` into account.

Reply:
I think `function_range` generates the range of a function, not it's domain? Also, from the PR of the previous logic: (#13335 and notably https://github.com/sympy/sympy/pull/13335#discussion_r140632892), it looks the different cases were introduced to work around https://github.com/sympy/sympy/issues/12527. `Intersection(solveset(1/f, symbol), domain)` is preferable to as it returns the correct answer, but if an expression contains an `Abs`, solveset will not work with a complex domain, and if you have a power of 2 a infinite recursion happens (you can see this by trying the test with `Intersection(solveset(1/f, symbol), domain)` everywhere with the old code).
Anyway, I don't think we'll have to use these cases in the new logic as #13335 isn't a problem as we're not going to be dealing with infinities which can be eliminated later on.

Reply:
Also, I'm assuming that `continuous_domain` is dealing with "non-absolute continuity", as this is the current behaviour. As you've said, it may be better if the documentation is clear up whether this is the case or not

Reply:
> generates the range of a function, not it's domain?

doh - myopia